src/commands/check.rs

```diff
@@ -119,9 +124,60 @@ pub async fn check(args: CheckArgs) -> anyhow::Result<()> {
         );
     }
 
-    let remote_file = Url::parse(&file).is_ok();
+    let remote_file = Url::parse(file).is_ok();
+
+    if !exceptions.is_empty() {
+        let analysis_rules: HashSet<String> = analysis::rules()
+            .iter()
+            .map(|r| r.id().to_lowercase())
+            .collect();
+        let lint_rules: HashSet<String> = lint::rules()
+            .iter()
+            .map(|r| r.id().to_lowercase())
+            .collect();
+        let all_rules: HashSet<_> = analysis_rules.union(&lint_rules).collect();
+
+        let unknown_exceptions: Vec<&str> = exceptions
+            .iter()
+            .filter(|rule| !all_rules.contains(&rule.to_lowercase()))
+            .map(|rule| rule.as_str())
+            .collect();
+
+        if !unknown_exceptions.is_empty() {
+            let (config, writer) =
+                get_display_config(args.common.report_mode, args.common.no_color);
+            let mut w_lock = writer.lock();
+            let files: SimpleFiles<String, String> = SimpleFiles::new();
+
+            let message = if unknown_exceptions.len() == 1 {
+                format!(
+                    "ignoring unknown rule provided via --except: '{}'",
+                    unknown_exceptions[0]
+                )
+            } else {
+                format!(
+                    "ignoring unknown rules provided via --except: '{}'",
+                    unknown_exceptions
+                        .iter()
+                        .map(|r| format!("`{r}`"))
+                        .collect::<Vec<_>>()
+                        .join(", ")
+                )
+            };
+
+            let warning = codespan_reporting::diagnostic::Diagnostic::warning()
+                .with_message(message)
+                .with_notes(vec![
+                    "Rule names are case-insensitive".to_string(),
```

Comment:
IMO this doesn't  belong as part of the diagnostic. I'd remove this (but is this specified in any of the doc strings? We should make that clear in the `clap` doc strings if it isn't)

Reply:
I thought since `explain` can take case insensitive rules, so check could also take one

Reply:
just checked with my local `sprocket` install, and it appears we do require exact capitalization for `--except` args, although I think you're right that we should probably relax that. I'll create an issue

Reply:
https://github.com/stjude-rust-labs/sprocket/issues/95

Reply:
@dead8309 remove this line about case insensitivity (which isn't true)